packages/repository/src/repositories/kv.repository.bridge.ts

```diff
@@ -0,0 +1,96 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import * as legacy from 'loopback-datasource-juggler';
+
+import {Options, DataObject} from '../common-types';
+import {Model} from '../model';
+
+import {KVRepository, AsyncIterator, KVFilter} from './kv.repository';
+
+import {juggler, ensurePromise} from './legacy-juggler-bridge';
+
+/**
+ * Polyfill for Symbol.asyncIterator
+ * See https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-3.html
+ */
+// tslint:disable-next-line:no-any
+if (!(Symbol as any).asyncIterator) {
+  // tslint:disable-next-line:no-any
+  (Symbol as any).asyncIterator = Symbol.for('Symbol.asyncIterator');
+}
+
+/**
+ * An implementation of KVRepository based on loopback-datasource-juggler
+ */
+export class DefaultKVRepository<T extends Model> implements KVRepository<T> {
+  /**
+   * A legacy KeyValueModel class
+   */
+  kvModelClass: typeof juggler.KeyValueModel;
+
+  /**
+   * Construct a KVRepository with a legacy DataSource
+   * @param ds Legacy DataSource
+   */
+  constructor(ds: juggler.DataSource) {
+    // KVModel class is placeholder to receive methods from KeyValueAccessObject
+    // through mixin
+    this.kvModelClass = ds.createModel<typeof juggler.KeyValueModel>(
+      '_kvModel',
+    );
+  }
+
+  delete(key: string, options?: Options): Promise<void> {
+    return ensurePromise(this.kvModelClass.delete(key, options));
+  }
+
+  deleteAll(options?: Options): Promise<void> {
+    return ensurePromise(this.kvModelClass.deleteAll(options));
+  }
+
+  get(key: string, options?: Options): Promise<T> {
+    const val = this.kvModelClass.get(key, options) as legacy.PromiseOrVoid<T>;
+    return ensurePromise<T>(val);
+  }
+
+  set(key: string, value: DataObject<T>, options?: Options): Promise<void> {
+    return ensurePromise<void>(this.kvModelClass.set(key, value, options));
+  }
+
+  expire(key: string, ttl: number, options?: Options): Promise<void> {
+    return ensurePromise<void>(this.kvModelClass.expire(key, ttl, options));
+  }
+
+  ttl(key: string, options?: Options): Promise<number> {
+    return ensurePromise<number>(this.kvModelClass.ttl(key, options));
+  }
+
+  keys(filter?: KVFilter, options?: Options): Promise<string[]> {
```

Comment:
In the thread https://github.com/strongloop/loopback-next/pull/1539#discussion_r203004909, you said you agree to drop `iterateKeys` and rework `keys` to return an async iterator. However, I see the old API is still present. Perhaps an oversight?

Reply:
Fixed